tests/unit/utils/merge-deep-test.js

```diff
@@ -35,4 +35,32 @@ module('Unit | Utility | merge deep', () => {
 
     assert.deepEqual(value, { company: { employees: ['Jull', 'Olafur']} }, 'has right employees');
   });
+
+  test('it works with unsafe properties', async function(assert) {
```

Comment:
What do you mean by "unsafe" here?  Just curious b/c we also use this term to reference prototype poisoning in this codebase

Reply:
I wasn't sure about the term, I used exactly the same term I found in the source code. I didn't find tests which were testing that branch of logic, so I've tried to "reverse engineer" the code to make tests go through that branch.